File: src/main/java/com/study/platform/domain/notification/model/NotificationEvent.java
package com.study.platform.domain.notification.model;

import java.util.UUID;

public record NotificationEvent(
        UUID receiverId,
        NotificationType type,
        String message,
        UUID targetId,
        int retryCount
) {
    public NotificationEvent withRetry() {
        return new NotificationEvent(receiverId, type, message, targetId, retryCount + 1);
    }
}

File: domain/src/main/java/com/study/platform/domain/notification/model/NotificationEvent.kt
package com.study.platform.domain.notification.model

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonProperty
import java.util.UUID

data class NotificationEvent @JsonCreator constructor(
    @JsonProperty("receiverId") val receiverId: UUID,
    @JsonProperty("type") val type: NotificationType,
    @JsonProperty("message") val message: String,
    @JsonProperty("targetId") val targetId: UUID?,
    @JsonProperty("retryCount") val retryCount: Int
) {
    fun withRetry(): NotificationEvent = copy(retryCount = retryCount + 1)
}
